src/tests/ftest/dfuse/daos_build.py

```diff
@@ -127,6 +143,11 @@ def run_build_test(self, cache_mode, intercept=False, run_on_vms=False):
         # scons commands which can both take a long time.
         build_time = 60
 
+        if il_lib is not None:
+            if il_lib == 'libpil4dfs.so':
+                # daos build takes ~4 hours in CI with tcp provider. 4~8 hours with verbs provider.
```

Comment:
Previous test of daos build with tcp provider normally took ~4 hours. The time decreased to ~2 hours after we introduced the flag to avoid daos_init() reenter. I still do not fully understand why the time could decrease so much. Maybe there are things changed in CI cluster.
I can run some local tests to see whether the time also decreased significantly.

Reply:
> The time decreased to ~2 hours after we introduced the flag to avoid daos_init() reenter

Wow.  If that a re-entry change then perhaps now it's not loading shared libs from dfuse?

Reply:
daos build with libpil4dfs finished in ~2 hours without issues in vm in two tests (build 27 and 28) since yesterdays. The speedup compared with previous ~4 hours probably comes from your PR on March 26, https://github.com/daos-stack/daos/pull/14049/files. "DAOS-15497 build: Only set LD_LIBRARY_PATH in scons build where required"
Avoiding daos_init() should be not relevant to the speedup.

Reply:
This makes more sense.  If there test is two hours on vms vs two hours on hardware then it should run on vms as the run-queues there are (typically) a lot shorter.